src/openforms/forms/api/serializers/form_definition.py

```diff
@@ -131,6 +132,20 @@ def to_representation(self, instance):
 
         return representation
 
+    def create(self, validated_data):
+        if configuration := validated_data.get("configuration"):
+            for component in configuration.get("components", []):
+                sanitize_component(component)
```

Comment:
this doesn't properly recurse for nested components - try writing a unit test with a fieldset with nested components.

We have `iter_components` for this :)